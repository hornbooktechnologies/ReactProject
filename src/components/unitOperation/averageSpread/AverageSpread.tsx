import React from "react";
import { useTranslation } from "react-i18next";
import { NegOperationAvgSpreadModel } from "../../../models/unitOperation/NegOperationAvgSpreadModel.ts";
import AverageSpreadCard from "./AverageSpreadCard";

const AverageSpread = ({ negOperationAvrgSpread }: NegOperationAvgSpreadModel) => {
  const { i18n } = useTranslation();
  return (
    <>
      <div>
        <h5 className="mb-24 mt-24 font-18 uk-text-medium">
          {i18n.t("UNIT_OPERATION.AVERAGE_SPREAD")}
        </h5>
        <div
          className="uk-grid uk-child-width-1-3@s uk-child-width-1-3@m"
          uk-grid="true"
        >
          {negOperationAvrgSpread?.map((obj, index) => {
            return (
              <React.Fragment key={index}>
                <AverageSpreadCard {...obj} />
              </React.Fragment>
            );
          })}
        </div>
      </div>
    </>
  );
};

export default AverageSpread;
