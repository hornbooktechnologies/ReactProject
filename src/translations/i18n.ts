import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
