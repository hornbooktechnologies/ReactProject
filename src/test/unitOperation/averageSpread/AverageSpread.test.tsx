import React from "react";
import renderer from "react-test-renderer";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";
import { NegOperationAvgSpreadModel } from "../../../models/unitOperation/NegOperationAvgSpreadModel.ts";
import AverageSpread from "../../../components/unitOperation/averageSpread/AverageSpread";

describe("Component: AverageSpread", () => {
  it("AverageSpread renders correctly", () => {
   
    const mockData = [
      {
          title: "Annual",
          Suffix: "円/kWh",
          plan: {
              header: "Plan",
              value: "0.00",
              previousValue: "0.00"
          },
          actual_forcast: {
              header: "Actual/Forecast",
              value: "-4.03",
              previousValue: "-4.03"
          }
      },
      {
          title: "4/1 ~ Present",
          Suffix: "円/kWh",
          plan: {
              header: "Plan",
              value: "0.00",
              previousValue: "0.00"
          },
          actual_forcast: {
              header: "Actual/Forecast",
              value: "-4.03",
              previousValue: "-4.03"
          }
      },
      {
          title: "Present ~ 3/31",
          Suffix: "円/kWh",
          plan: {
              header: "Plan",
              value: "0.00",
              previousValue: "0.00"
          },
          actual_forcast: {
              header: "Actual/Forecast",
              value: "0.00",
              previousValue: "0.00"
          }
      }
  ]
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <AverageSpread {...mockData} />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
