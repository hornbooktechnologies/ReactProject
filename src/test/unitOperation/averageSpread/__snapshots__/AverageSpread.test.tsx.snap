// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AverageSpread AverageSpread renders correctly 1`] = `
<div>
  <h5
    className="mb-24 mt-24 font-18 uk-text-medium"
  >
    UNIT_OPERATION.AVERAGE_SPREAD
  </h5>
  <div
    className="uk-grid uk-child-width-1-3@s uk-child-width-1-3@m"
    uk-grid="true"
  />
</div>
`;
